doc/man3/STORE_LOADER.pod

```diff
@@ -0,0 +1,192 @@
+ int STORE_LOADER_set_eof(STORE_LOADER *store_loader,
+                            STORE_eof_fn store_eof_function);
+ typedef int (*STORE_close_fn)(STORE_LOADER_CTX *ctx);
+ int STORE_LOADER_set_close(STORE_LOADER *store_loader,
+                            STORE_close_fn store_close_function);
+ void STORE_LOADER_free(STORE_LOADER *store_loader);
+ 
+ int STORE_register_loader(STORE_LOADER *loader);
+ STORE_LOADER *STORE_unregister_loader(const char *scheme);
+
+=head1 DESCRIPTION
+
+These functions help applications and engines to create loaders for
+schemes they support.
+
+=head2 Types
+
+B<STORE_LOADER> is the type to hold a loader.  It contains a scheme
+and the functions needed to implement STORE_open(), STORE_eof(),
+STORE_load() and STORE_close() for this scheme.
+
+B<STORE_LOADER_CTX> is an type template, to be defined by each loader
+using B<struct store_loader_ctx_st { ... }>.
+
+B<STORE_open_fn>, B<STORE_ctrl_fn>, B<STORE_load_fn>, B<STORE_eof_fn>
+and B<STORE_close_fn> are the function pointer types used within a
+STORE loader.  The functions pointed at define the functionality of
+the given loader.
+
+=over 4
+
+=item B<STORE_open_fn>
+
+This function takes a URI and expected to interpret it in the best
+manner possible according to the scheme the loader implements, it also
+takes a B<UI_METHOD> and associated data, to be used any time
+something needs to be prompter for.  Furthermore, this function is
+expected to initialize what needs to be initialized, to create a
+privata data store (B<STORE_LOADER_CTX>, see above), and to return
+it.  If something goes wrong, this function is expected to return
+NULL.
+
+=item B<STORE_ctrl_fn>
+
+This function takes a B<STORE_LOADER_CTX> pointer, a command number
+B<cmd> and a B<va_list> B<args> and is used to manipulate loader
+specific parameters.
+
+=begin comment
+
+Globally known command numbers are documented in L<STORE_ctrl(3)>,
+along with what B<args> are expected with each of them.
+
+=end comment
+
+Loader specific command numbers must begin at B<STORE_C_CUSTOM_START>.
+Any number below that is reserved for future globally known command
+numbers.
+
+This function is expected to return 1 on success, 0 on error.
+
+=item B<STORE_load_fn>
+
+This function takes a B<STORE_LOADER_CTX> pointer and a B<UI_METHOD>
+with associated data.  It's expected to load the next available data,
+mold it into a data structure that can be wrapped in a B<STORE_INFO>
+using one of the L<STORE_INFO(3)> functions.  If not more data is
+available, this function is expected to call STORE_INFO_new_ENDOFDATA()
+and to return its result.  Should an error occur, this function is
+expected to return NULL.
+
+Note that this function retrives I<one> data item only.
+
+=item B<STORE_eof_fn>
+
+This function takes a B<STORE_LOADER_CTX> pointer and is expected to
+return 1 if it expects that more data can be loaded from the store, or
+when it doesn't know.  It's expected to return 0 when it knows for
```

Comment:
Not only that, but that documentation is incorrect.  That's not how `STORE_eof()` or `loader_file.c`'s `file_eof` behave.